File: 백준/Silver/32713. 숫자 POP/숫자 POP.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")

// .map(x => x.trim())


function solution([nk, arg]) {
  let digits = arg.split(' ').map(Number)
  let [n, k] = nk.split(' ').map(Number)
  let max = 0
  for (let i = 0; i < n; i++) {
    let current = digits[i]
    let use = k
    let count = 1
    for (let j = i + 1; j < n; j++) {
      if (current !== digits[j]) {
        if (use > 0) {
          use -= 1
        } else break
      } else count++
    }
    max = Math.max(max, count)
  }
  console.log(max)

}

solution(input)

File: 백준/Silver/32713. 숫자 POP/숫자 POP.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.util.*

fun main() {
    val os = System.getProperty("os.name").lowercase(Locale.getDefault())
    val args: List<String> = if (os.contains("linux")) {
        val br = BufferedReader(InputStreamReader(System.`in`))
        br.readLines()
    } else {
        val inputFile = File("src/main/kotlin/input.txt")
        inputFile.readLines()
    }

    val (n, k) = args[0].split(' ').map { it.toInt() }
    val digits = args[1].split(' ').map { it.toInt() }
    var max = 0
    for(i in 0 until n){
        val current = digits[i]
        var count = 1
        var use = k
        for(j in i + 1 until n){
            if(current != digits[j]){
                if(use > 0){
                    use--
                } else break
            } else count++
        }
        max = maxOf(max, count)
    }
    print(max)

}
